presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -682,12 +694,53 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
             tableProperties.put(ORC_BLOOM_FILTER_FPP_KEY, String.valueOf(getOrcBloomFilterFpp(tableMetadata.getProperties())));
         }
 
+        // AVRO specific properties
+        String avroSchemaUrl = getAvroSchemaUrl(tableMetadata.getProperties());
+        if (avroSchemaUrl != null) {
+            HiveStorageFormat hiveStorageFormat = getHiveStorageFormat(tableMetadata.getProperties());
+            if (hiveStorageFormat != HiveStorageFormat.AVRO) {
+                throw new PrestoException(INVALID_TABLE_PROPERTY, format("Cannot specify %s table property for storage format: %s", AVRO_SCHEMA_URL, hiveStorageFormat));
+            }
+            tableProperties.put(AVRO_SCHEMA_URL_KEY, validateAndNormalizeAvroSchemaUrl(avroSchemaUrl, hdfsContext));
+        }
+
         // Table comment property
         tableMetadata.getComment().ifPresent(value -> tableProperties.put(TABLE_COMMENT, value));
 
         return tableProperties.build();
     }
 
+    private String validateAndNormalizeAvroSchemaUrl(String url, HdfsContext context)
+    {
+        try {
+            new URL(url).openStream().close();
+            return url;
+        }
+        catch (MalformedURLException e) {
+            // try locally
+            if (new File(url).exists()) {
+                // hive needs url to have a protocol
+                url = "file:///" + url;
```

Comment:
This seems incorrect:

* A relative path like `foo/bar` will become an absolute path `file:///foo/bar`.
* An absolute path like `/foo/bar` will have four slashes `file:////foo/bar`. This probably works, but is incorrect according to RFC 8089. It should either have one slash (no authority) or three slashes (empty authority).

It seems safer to construct it like this, which guarantees the format is correct:
```java
return new File(url).toURI().toString();